packages/server/customer-os-api/service/service_line_item_service.go

```diff
@@ -0,0 +1,167 @@
+	return &serviceLineItemService{
+		log:          log,
+		repositories: repositories,
+		grpcClients:  grpcClients,
+		services:     services,
+	}
+}
+
+type ServiceLineItemCreateData struct {
+	ServiceLineItemEntity *entity.ServiceLineItemEntity
+	ContractId            string
+	ExternalReference     *entity.ExternalSystemEntity
+	Source                entity.DataSource
+	AppSource             string
+}
+
+func (s *serviceLineItemService) Create(ctx context.Context, serviceLineItemDetails *ServiceLineItemCreateData) (string, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "ServiceLineItemService.Create")
+	defer span.Finish()
+	tracing.SetDefaultServiceSpanTags(ctx, span)
+	span.LogFields(log.Object("serviceLineItemDetails", serviceLineItemDetails))
+
+	if serviceLineItemDetails.ServiceLineItemEntity == nil {
+		err := fmt.Errorf("service line item entity is nil")
+		tracing.TraceErr(span, err)
+		return "", err
+	}
+
+	serviceLineItemId, err := s.createServiceLineItemWithEvents(ctx, serviceLineItemDetails)
+	if err != nil {
+		tracing.TraceErr(span, err)
+		s.log.Errorf("Error from events processing: %s", err.Error())
+		return "", err
+	}
+
+	span.LogFields(log.String("output - createdServiceLineItemId", serviceLineItemId))
+	return serviceLineItemId, nil
+}
+
+func (s *serviceLineItemService) createServiceLineItemWithEvents(ctx context.Context, serviceLineItemDetails *ServiceLineItemCreateData) (string, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "ServiceLineItemService.createServiceLineItemWithEvents")
+	defer span.Finish()
+	tracing.SetDefaultServiceSpanTags(ctx, span)
+
+	createServiceLineItemRequest := servicelineitempb.CreateServiceLineItemGrpcRequest{
+		Tenant:         common.GetTenantFromContext(ctx),
+		ContractId:     serviceLineItemDetails.ContractId,
+		Name:           serviceLineItemDetails.ServiceLineItemEntity.Name,
+		Quantity:       int64(serviceLineItemDetails.ServiceLineItemEntity.Quantity),
+		Price:          float32(serviceLineItemDetails.ServiceLineItemEntity.Price),
+		LoggedInUserId: common.GetUserIdFromContext(ctx),
+		SourceFields: &commonpb.SourceFields{
+			Source:    string(serviceLineItemDetails.Source),
+			AppSource: utils.StringFirstNonEmpty(serviceLineItemDetails.AppSource, constants.AppSourceCustomerOsApi),
+		},
+	}
+
+	switch serviceLineItemDetails.ServiceLineItemEntity.Billed {
+	case entity.BilledTypeMonthly:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_MONTHLY_BILLED
+	case entity.BilledTypeAnnually:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_ANNUALLY_BILLED
+	case entity.BilledTypeOnce:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_ONCE_BILLED
+	default:
+		//TODO add to service_line_item proto billed type NONE_BILLED and use it as default here
```

Comment:
there is no none option so far, there should be an option mandatory sent on create